tools/gnur-runtime/src/main/java/org/renjin/gnur/api/Rinternals.java

```diff
@@ -456,7 +456,12 @@ public static void SET_ATTRIB(SEXP x, SEXP v) {
   }
 
   public static void DUPLICATE_ATTRIB(SEXP to, SEXP from) {
-    throw new UnimplementedGnuApiMethod("DUPLICATE_ATTRIB");
+    AbstractSEXP abstractSEXP = (AbstractSEXP) to;
+    if (Types.isS4(from)) {
+      abstractSEXP.unsafeSetAttributes(from.getAttributes().copy().setS4(true));
```

Comment:
Renjin actually encodes the S4 flag in the AttributeMap object, so all that should be neccessary is

```.java
abstractSEXP.unsafeSetAttributes(from.getAttributes());
```
There is one thing that we may run into: I would worry that the caller of this function will assume that they can modify the attributes on `to` in place, despite the fact that Renjin sets the `NAMED` bit to true. If the caller does do this, it will affect the attributes on both `to` and `from` because here they are shared. (Note that the AttributeMap.copy method only creates a shallow copy of the AttributeMap). 

@psolaimani I believe you were also working on an implementation of this method. Is a deep copy neccessary here?

Reply:
From what I understood reading the code the `AttributeMap.copy()` holds a copy of the `HashMap` and a reference to read-only `StringVector`s and `IntVector`.